src/pybind/mgr/dashboard/frontend/src/app/shared/api/prometheus.service.ts

```diff
@@ -79,6 +80,10 @@ export class PrometheusService {
     return this.http.get<AlertmanagerAlert[]>(this.baseURL, { params });
   }
 
+  getGroupedAlerts() {
+    return this.http.get<GroupAlertmanagerAlert[]>(`${this.baseURL}/alertgroup`);
```

Comment:
I think we can just do the update in existing api getAlerts()

As that I see using `cluster_filter` and it might be breaking there. So we need to update existing one:

```
 getAlerts(clusterFilteredAlerts = false, params = {}): Observable<AlertmanagerAlert[]> {
    params['cluster_filter'] = clusterFilteredAlerts;
    return this.http.get<AlertmanagerAlert[]>(this.baseURL, { params });
  }
  ```
  
  as well update in `promethues.py` to accept cluster filter same as in /alert
  
  ```
  @RESTController.Collection(method='GET', path='/alertgroup')
    def get_alertgroup(self, **params):
        return self.alert_proxy('GET', '/alerts/groups', params)
```

cc @nizamial09 @aaSharma14 ^^^ correct me if I am wrong

Reply:
yup, this needs the `cluster_filter` and I don't know if we need to do something in the prometheus for this. @cloudbehl

Reply:
I think we need to test it out. @aaSharma14 should be able to help here.